### PR TITLE
Add filterPrecursorMaxIntensity and filterPrecursorIsotopes functions

### PR DESCRIPTION
**Pull request by AharoniLab**
This PR adds the filterPrecursorMaxIntensity (issue https://github.com/rformassspectrometry/Spectra/issues/280) and filterPrecursorIsotopes (issue https://github.com/rformassspectrometry/Spectra/issues/281) functions along with documentation and unit tests.

### CODE REVIEW

**Review by lgatto — COMMENTED** [review]
Thank you. 

If the precursor intensities aren't available to run `filterPrecursorMaxIntensity()`, they can be computed with `estimatePrecursorIntensity()`. Have you compared the results the using the latter for data that do have `precursorIntensity()`? This would be a useful illustration/example and/or unit test.

File: R/Spectra-functions.R [hunk 1]
```diff
@@ -813,6 +814,67 @@ deisotopeSpectra <-
...
 }
+
+#' @rdname Spectra
+#'
+#' @author Nir Shahaf
+#'
+#' @export
+filterPrecursorMaxIntensity <- function(x, tolerance = 0, ppm = 20) {
+    pmz <- precursorMz(x)
+    pmi <- precursorIntensity(x)
```
[the remark this answers is not in the document]
Comment (jorainer) [reply R/Spectra-functions.R:889]: We're now checking if the precursor intensities are `NA` and eventually pointing the user to the `estimatePrecursorIntensity` function.

**Comment by jorainer** [comment]
@lgatto , we addressed the comments. An example for `estimatePrecursorIntensities` comparing them against those defined by the manufacturer's software is given in the example section for that function. Also, a unit test was already in place before.

Note that the failing unit test for linux is due to a bug/problem in the `remotes` package. This will be fixed in the next release of the `remotes` package.

**Comment by jorainer** [comment]
After discussion with @lgatto we agreed to add the `filterPrecursorMaxIntensity` function. For the `filterPrecursorIsotopes` some more discussion is needed whether it would not make more sense to add that to e.g. `xcms` or `MetaboAnnotation` instead. That depends how *common* such data (with precursor isotopes) actually is.

**Comment by AharoniLab** [comment]
Hi Johannes,


I cannot follow the logic of splitting these very related function between packages. I can only say that the phenomena of precursor (and fragment) isotopic ions occurs always in DIA-MS methods and quite often also in DDA data (depending on instrument type, MS method settings etc.). Should a "filter isotopes" function be added to 'xcms'? Absolutely, but that in my view should be decoupled from the need for such a function in annotating Spectra because there can be other application workflows.


Nir

________________________________

**Comment by jorainer** [comment]
For whatever reason this PR is not picking up the recent commits that fix the merge conflicts. I will close this PR and open a new one instead.

**Pull request closed without merging**